File: src/BLite.Client/BLiteClientOptions.cs
// BLite.Client — BLiteClientOptions
// Copyright (C) 2026 Luca Fabbri — AGPL-3.0

namespace BLite.Client;

/// <summary>
/// Configuration for connecting to a remote BLite Server.
/// </summary>
public sealed class BLiteClientOptions
{
    /// <summary>Server hostname or IP. Default: localhost.</summary>
    public string Host { get; set; } = "localhost";

    /// <summary>Server gRPC port. Default: 2626.</summary>
    public int Port { get; set; } = 2626;

    /// <summary>API key sent as <c>x-api-key</c> metadata on every call.</summary>
    public string ApiKey { get; set; } = string.Empty;

    /// <summary>
    /// Whether to use TLS. Default: <c>true</c>.
    /// Set to <c>false</c> only for local development / testing.
    /// </summary>
    public bool UseTls { get; set; } = true;

    /// <summary>
    /// Optional: override the full address (e.g. "https://myserver:2626").
    /// When set, <see cref="Host"/>, <see cref="Port"/> and <see cref="UseTls"/>
    /// are ignored.
    /// </summary>
    public string? Address { get; set; }

    internal string ResolvedAddress =>
        Address ?? $"{(UseTls ? "https" : "http")}://{Host}:{Port}";
}

File: src/BLite.Client.Java/src/main/java/io/blite/client/BLiteClientOptions.java
// BLite.Client.Java — BLiteClientOptions
// Copyright (C) 2026 Luca Fabbri — AGPL-3.0

package io.blite.client;

/** Connection and authentication options for {@link BLiteClient}. */
public final class BLiteClientOptions {

    private String  host   = "localhost";
    private int     port   = 2626;
    private String  apiKey;
    private boolean useTls = true;

    public BLiteClientOptions() {}

    public BLiteClientOptions(String host, int port, String apiKey, boolean useTls) {
        this.host   = host;
        this.port   = port;
        this.apiKey = apiKey;
        this.useTls = useTls;
    }

    public String  getHost()   { return host; }
    public int     getPort()   { return port; }
    public String  getApiKey() { return apiKey; }
    public boolean isUseTls()  { return useTls; }

    public BLiteClientOptions host(String host)     { this.host = host;     return this; }
    public BLiteClientOptions port(int port)        { this.port = port;     return this; }
    public BLiteClientOptions apiKey(String apiKey) { this.apiKey = apiKey; return this; }
    public BLiteClientOptions useTls(boolean tls)   { this.useTls = tls;   return this; }

    /** Convenience: insecure localhost connection (for dev/testing). */
    public static BLiteClientOptions local(String apiKey) {
        return new BLiteClientOptions("localhost", 2626, apiKey, false);
    }
}
